xarray/core/variable.py

```diff
@@ -936,6 +936,48 @@ def shift(self, **shifts):
             result = result._shift_one_dim(dim, count)
         return result
 
+    def pad_with_fill_value(self, fill_value=None, **pad_widths):
+        """
+        Return a new Variable with paddings.
+
+        Parameters
+        ----------
+        **pad_width: keyword arguments of the form {dim: (before, after)}
+            Number of values padded to the edges of each dimension.
+        """
+        if fill_value is None:
```

Comment:
Use `dtypes.NA` as the default value here -- that's a sentinel value specifically designed for this use case. Otherwise you can't pad with `None` as a fill value.